testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/VaultUtils.java

```diff
@@ -1,34 +1,60 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package org.keycloak.testsuite.util;
 
 
 import org.keycloak.testsuite.arquillian.AuthServerTestEnricher;
 import org.keycloak.testsuite.arquillian.SuiteContext;
+import org.keycloak.testsuite.arquillian.annotation.EnableVault;
 import org.wildfly.extras.creaper.core.online.CliException;
 import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
 
 import java.io.IOException;
 import java.util.concurrent.TimeoutException;
 
+import static org.keycloak.testsuite.arquillian.annotation.EnableVault.PROVIDER_ID.ELYTRON_CS_KEYSTORE;
+import static org.keycloak.testsuite.arquillian.annotation.EnableVault.PROVIDER_ID.PLAINTEXT;
+
 
 /**
  * @author mhajas
  */
 public class VaultUtils {
 
-    public static void enableVault(SuiteContext suiteContext) throws IOException, CliException, TimeoutException, InterruptedException {
+    public static void enableVault(SuiteContext suiteContext, EnableVault.PROVIDER_ID provider) throws IOException, CliException, TimeoutException, InterruptedException {
         if (suiteContext.getAuthServerInfo().isUndertow()) {
-            System.setProperty("keycloak.vault.plaintext.provider.enabled", "true");
+            System.setProperty("keycloak.vault." + provider.getName() + ".provider.enabled", "true");
         } else {
             OnlineManagementClient client = AuthServerTestEnricher.getManagementClient();
+            // add all available providers, enabling only the one used in the test - could also enable all and set the default provider to be the one used in the test.
             client.execute("/subsystem=keycloak-server/spi=vault/:add");
-            client.execute("/subsystem=keycloak-server/spi=vault/provider=files-plaintext/:add(enabled=true,properties={dir => \"${jboss.home.dir}/standalone/configuration/vault\"})");
+            client.execute("/subsystem=keycloak-server/spi=vault/provider=elytron-cs-keystore/:add(enabled=" + (provider == ELYTRON_CS_KEYSTORE) +
```

Comment:
This seems a bit fragile. Is there any better, more declarative and object oriented way to implement this? If not - I guess this one is fine.

Reply:
@slaskawi yeah, it was not the best I could come up with. I changed it a little, moving the installation command to each provider in the enum and only installing/configuring the one that was selected, setting it also as the default provider. It looks a lot better now but if you have any other suggestions fire away :)